WebApi/WebApiTest/RepositoryTest/PostRepositoryTest.cs

```diff
@@ -0,0 +1,154 @@
+                SeedPost(dbContext);
+                int expectedID = 1;
+                var expected = dbContext.Posts.Where(x => x.PostID == expectedID).FirstOrDefault();
+
+                var cls = new PostRepository(dbContext);
+                var actual = cls.GetPostByIdAsync(expectedID);
+
+                Assert.True(actual != null);
+                Assert.Equal(expected.PostID, actual.Result.PostID);
+                Assert.Equal(expected.UserID, actual.Result.UserID);
+                Assert.Equal(expected.CategoryID, actual.Result.CategoryID);
+                Assert.Equal(expected.Date, actual.Result.Date);
+                Assert.Equal(expected.Title, actual.Result.Title);
+                Assert.Equal(expected.Content, actual.Result.Content);
+                Assert.Equal(expected.Localization, actual.Result.Localization);
+                Assert.Equal(expected.ShopName, actual.Result.ShopName);
+                Assert.Equal(expected.IsPromoted, actual.Result.IsPromoted);
+
+            }
+        }
+
+        [Fact]
+        public void GetPostByIdAsync_InvalidCall_NoId()
+        {
+            var options = new DbContextOptionsBuilder<DatabaseContext>()
+                .UseInMemoryDatabase(databaseName: "GetPostByIdAsync_InvalidCall_NoId").Options;
+
+            using (var dbContext = new DatabaseContext(options))
+            {
+                SeedPost(dbContext);
+                int expectedID = 0;
+                var expected = dbContext.Posts.Where(x => x.PostID == expectedID).FirstOrDefault();
+
+                var cls = new PostRepository(dbContext);
+                var actual = cls.GetPostByIdAsync(expectedID);
+
+                Assert.True(actual != null);
+                Assert.True(actual.Result == null);
+            }
+        }
+
+        [Fact]
+        public void EditPostAsync_ValidCall()
+        {
+            var options = new DbContextOptionsBuilder<DatabaseContext>()
+                .UseInMemoryDatabase(databaseName: "EditPostAsync_ValidCall").Options;
+
+            var editBody = new PostEditDTO
+            {
+                title = "newtitle",
+                content = "newcontent",
+                category = 5,
+                dateTime = DateTime.UtcNow,
+                isPromoted = true
+            };
+
+            using (var dbContext = new DatabaseContext(options))
+            {
+                SeedPost(dbContext);
+
+                int changedPostID = 1;
+
+
+                var expected = dbContext.Posts.Where(x => x.PostID == changedPostID).FirstOrDefault();
+                expected.Title = editBody.title;
+                expected.Content = editBody.content;
+                expected.CategoryID = editBody.category;
+                expected.Date = editBody.dateTime;
+                expected.IsPromoted = editBody.isPromoted;
+
+
+
+                var cls = new PostRepository(dbContext);
+                var actual = cls.EditPostAsync(changedPostID, editBody);
```

Comment:
Ciekawe że pozawala w ramach jednego using jednocześnie dodać i edytować, bo pamiętam że w RepositoryTest nie pozwalało tak zrobić, ale jak działa to jest oki